c2corg_api/views/image.py

```diff
@@ -19,10 +29,42 @@ def collection_get(self):
     def get(self):
         return self._get(Image, schema_image)
 
-    @restricted_json_view(schema=schema_image)
+    @restricted_json_view(
+            schema=schema_image, validators=validate_image_create)
     def collection_post(self):
         return self._collection_post(schema_image)
 
-    @restricted_json_view(schema=schema_update_image, validators=validate_id)
+    @restricted_json_view(
+            schema=schema_update_image,
+            validators=[validate_id, validate_image_update])
     def put(self):
+        if not self.request.has_permission('moderator'):
+            image_id = self.request.validated['id']
+            image = DBSession.query(Image).get(image_id)
+            if image is None:
+                raise HTTPNotFound('No image found for id %d' % image_id)
+            if image.image_type == 'collaborative':
+                image_type = self.request.validated['document']['image_type']
+                if image_type != image.image_type:
+                    raise HTTPBadRequest(
+                        'Image type cannot be changed for collaborative images'
+                    )
+            # personal images should only be modifiable by
+            # their creator and moderators
+            elif not self._is_creator(
+                 self.request.authenticated_userid,
+                 image_id):
+                raise HTTPForbidden('No permission to change this image')
         return self._put(Image, schema_image)
+
+    def _is_creator(self, user_id, image_id):
+        """Check if the user with the given id has permission to change a
+        personal image (is the author).
+        """
+        initial_version = DBSession.query(DocumentVersion) \
```

Comment:
I don't think this can work: we need to get the older version whatever the language and then check that the user_ids match. Your query only checks that the user has created a version (which might not be the first one).